npm/src/db/redis.ts

```diff
@@ -36,26 +36,36 @@ class Redis implements DatabaseDriver {
     return null;
   }
 
-  async getAll(namespace: string): Promise<unknown[]> {
-    const keys = await this.client.sendCommand(['keys', namespace + ':*']);
+  async getAll(namespace: string, pageOffset: number, pageLimit: number): Promise<unknown[]> {
+    const offsetAndLimitValueCheck = !dbutils.isNumeric(pageOffset) && !dbutils.isNumeric(pageLimit);
+    let take = Number(offsetAndLimitValueCheck ? this.options.pageLimit : pageLimit);
+    const skip = Number(offsetAndLimitValueCheck ? 0 : pageOffset);
     const returnValue: string[] = [];
-    for (let i = 0; i < keys.length; i++) {
-      try {
-        if (this.client.get(keys[i])) {
-          const value = await this.client.get(keys[i]);
-          returnValue.push(JSON.parse(value));
-        }
-      } catch (error) {
-        console.error(error);
+    let count = 0;
+    take += skip;
+
+    for await (const key of this.client.scanIterator({
+      MATCH: dbutils.keyFromParts(namespace, '*'),
+      COUNT: take,
```

Comment:
@vishalboxyhq `take` can get very large, should set count to a sane maximum value.